### DISCUSSION TITLE
Load animation dynamically from a specific file and can apply the animation to the 3d model and can play the animation on the 3d model

### DISCUSSION THREAD

**Discussion by whereisnothing** [google/filament#5820] [General]
**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is.

**Describe the solution you'd like**
We hava a demand. We already hava a 3d model and can play any animations on it. But sometimes we want to play an animation on the 3d model from network or local file system. That is, load an animation from a file and apply the animation on to the 3d model, so we can play newly applied animations anytime.

**Describe alternatives you've considered**

**OS and backend**
Is this feature OS/backend specific? If so, explain why.

**Comment by bejado** [comment]
How are you loading the models and what type of animation (skeletal, morph targets, etc) do you need?

As far as Filament is concerned, you're free to invent your own animation file format depending on your needs. You could look at the [glTF 2.0 animation spec](https://registry.khronos.org/glTF/specs/2.0/glTF-2.0.html#animations) for inspiration.

**Reply by whereisnothing** [reply]
We need skeletal animation.Filament api has only Animator.applyAnimation, and the parameters transferred to the api is the already owned animation index,not a new animation.So could you add this api?

**Reply by August1996** [reply]
> We need skeletal animation.Filament api has only Animator.applyAnimation, and the parameters transferred to the api is the already owned animation index,not a new animation.So could you add this api?

Maybe you can play animations with another glb file, and then retarget the bones to you model.

**Reply by ninjz** [reply]
> Maybe you can play animations with another glb file, and then retarget the bones to you model.

@August1996 how would you go about doing that?

**Reply by hannojg** [reply]
If the bones are the same you can basically do this:https://github.com/google/filament/issues/7622
